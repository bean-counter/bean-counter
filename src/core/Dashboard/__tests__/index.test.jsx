import React from 'react'
import renderer from 'react-test-renderer'
import { DashboardComponent } from '../'

jest.mock('../../../common/InstitutionIcon', () => 'InstitutionIcon')

const groupedAccounts = {
  TD: { accountIds: ['1', '2'], balance: 20 },
  BMO: { accountIds: ['3'], balance: 10 }
}

const mochFormatCurrency = jest.fn().mockReturnValue(0)

describe('Dashboard', () => {
  describe('snapshot', () => {
    it('matches with no accounts', () => {
      const component = renderer.create((
        <DashboardComponent
          formatCurrency={mochFormatCurrency}
          groupedAccounts={{}}
          totalBalance={0}
          classes={{ }}
        />
      ))
      expect(component.toJSON()).toMatchSnapshot()
    })

    it('matches with a few accounts', () => {
      const component = renderer.create((
        <DashboardComponent
          formatCurrency={mochFormatCurrency}
          groupedAccounts={groupedAccounts}
          totalBalance={30}
          classes={{ }}
        />
      ))
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
