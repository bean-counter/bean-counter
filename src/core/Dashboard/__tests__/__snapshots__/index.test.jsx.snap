// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard snapshot matches with a few accounts 1`] = `
<div
  className="MuiGrid-container-1"
>
  <div
    className="MuiGrid-container-1 MuiGrid-justify-xs-flex-end-19"
  >
    <div
      className="MuiPaper-root-98 MuiPaper-elevation2-102 MuiPaper-rounded-99"
    >
      <p
        className="MuiTypography-root-125 MuiTypography-body1-134 MuiTypography-colorTextSecondary-158 MuiTypography-alignRight-149"
      >
        Balance
      </p>
      <h5
        className="MuiTypography-root-125 MuiTypography-h5-141 MuiTypography-alignRight-149"
      >
        0
      </h5>
      <div
        className="MuiPaper-root-98 MuiPaper-elevation2-102 MuiPaper-rounded-99"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-160"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root-98 MuiPaper-elevation2-102 MuiPaper-rounded-99"
  >
    <table
      className="MuiTable-root-169"
    >
      <tbody
        className="MuiTableBody-root-170"
      >
        <tr
          className="MuiTableRow-root-171"
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root-176 MuiTableCell-body-178"
            scope="row"
          >
            TD
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root-176 MuiTableCell-body-178 MuiTableCell-alignRight-186"
          >
            0
          </td>
        </tr>
        <tr
          className="MuiTableRow-root-171"
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root-176 MuiTableCell-body-178"
            scope="row"
          >
            BMO
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root-176 MuiTableCell-body-178 MuiTableCell-alignRight-186"
          >
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Dashboard snapshot matches with no accounts 1`] = `
<div
  className="MuiGrid-container-1"
>
  <div
    className="MuiGrid-container-1 MuiGrid-justify-xs-flex-end-19"
  >
    <div
      className="MuiPaper-root-98 MuiPaper-elevation2-102 MuiPaper-rounded-99"
    >
      <p
        className="MuiTypography-root-125 MuiTypography-body1-134 MuiTypography-colorTextSecondary-158 MuiTypography-alignRight-149"
      >
        Balance
      </p>
      <h5
        className="MuiTypography-root-125 MuiTypography-h5-141 MuiTypography-alignRight-149"
      >
        0
      </h5>
      <div
        className="MuiPaper-root-98 MuiPaper-elevation2-102 MuiPaper-rounded-99"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-160"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root-98 MuiPaper-elevation2-102 MuiPaper-rounded-99"
  >
    <span
      className="MuiTypography-root-125 MuiTypography-caption-135"
    >
      You don't have any accounts yet
    </span>
    <table
      className="MuiTable-root-169"
    >
      <tbody
        className="MuiTableBody-root-170"
      />
    </table>
  </div>
</div>
`;
